eng/tools/npm-helpers.ps1

```diff
@@ -0,0 +1,76 @@
+$EngPath = Resolve-Path "${PSScriptRoot}/.."
+$Engcommon = Join-Path $EngPath "common"
+$EngCommonScriptsPath = Join-Path $Engcommon "scripts"
+. (Join-Path $EngCommonScriptsPath common.ps1)
+
+function Get-PackageProperties($pkgArtifact, $workingDirectory)
+{
+  if ($GetPackageInfoFromPackageFileFn -and (Test-Path "Function:$GetPackageInfoFromPackageFileFn"))
+  {
+      return &$GetPackageInfoFromPackageFileFn -pkg $packageArtifact -workingDirectory $workingDirectory
+  }
+  else
+  {
+    Write-Error "The function for '$GetPackageInfoFromPackageFileFn' was not found.`
+    Make sure it is present in eng/scripts/Language-Settings.ps1 and referenced in eng/common/scripts/common.ps1.`
+    See https://github.com/Azure/azure-sdk-tools/blob/master/doc/common/common_engsys.md#code-structure"
+    return $null
+  }
+}
+
+function Get-LatestVersionInfoFromNpm($packageName)
+{
+  $latestVersion = npm show $packageName@latest version
```

Comment:
We have some other npm rest calls at https://github.com/Azure/azure-sdk-for-js/blob/master/eng/scripts/Language-Settings.ps1#L281 we should see if we should use a similar mechanism or if we should use the npm tool directly for these.

Reply:
My intention is to have all npm specific calls within this npm helper script. But difference you have mentioned is REST v/s tool and one benefit I can think of is to hide errors when package is not even present. I can't think of any other difference in using REST.

Reply:
If there isn't much of a different I think I would lean toward rest because you will generally get better structured data and better errors.

Reply:
Done.